### ISSUE TITLE
draggging window crash desktop environnement

### ISSUE THREAD

**Issue by pwouik** [lapce/lapce#2651]
## Lapce Version

master branch build(f72401629787cea3a0cd45f2d62dc138b464f345)

## System information
archlinux KDE Plasma 5.27.8 x11

## Describe the bug
dragging/resizing the window cause it to disappear and sometimes crash plasma

**Comment by pwouik** [comment]
journalctl -r:
```oct. 24 22:19:07 archlinux systemd-coredump[62861]: [🡕] Process 578 (Xorg) of user 0 dumped core.

                                                    Stack trace of thread 578:
                                                    #0  0x00007f5f9f75983c n/a (libc.so.6 + 0x8e83c)
                                                    #1  0x00007f5f9f709668 raise (libc.so.6 + 0x3e668)
                                                    #2  0x00007f5f9f6f14b8 abort (libc.so.6 + 0x264b8)
                                                    #3  0x0000560aee7ee5a0 OsAbort (Xorg + 0x1535a0)
                                                    #4  0x0000560aee7efa0b FatalError (Xorg + 0x154a0b)
                                                    #5  0x0000560aee7f6516 n/a (Xorg + 0x15b516)
                                                    #6  0x00007f5f9f709710 n/a (libc.so.6 + 0x3e710)
                                                    #7  0x00007f5f9f75983c n/a (libc.so.6 + 0x8e83c)
                                                    #8  0x00007f5f9f709668 raise (libc.so.6 + 0x3e668)
                                                    #9  0x00007f5f9f6f14b8 abort (libc.so.6 + 0x264b8)
                                                    #10 0x00007f5f9c6a4b49 n/a (iris_dri.so + 0x9bb49)
                                                    #11 0x00007f5f9d46d377 n/a (iris_dri.so + 0xe64377)
                                                    #12 0x00007f5f9cc13e8f n/a (iris_dri.so + 0x60ae8f)
                                                    #13 0x00007f5f9c8b6486 n/a (iris_dri.so + 0x2ad486)
                                                    #14 0x0000560aee77ad96 n/a (Xorg + 0xdfd96)
                                                    #15 0x0000560aee77d11d n/a (Xorg + 0xe211d)
                                                    #16 0x0000560aee77bd85 n/a (Xorg + 0xe0d85)
                                                    #17 0x0000560aee6d9eae n/a (Xorg + 0x3eeae)
                                                    #18 0x00007f5f9f6f2cd0 n/a (libc.so.6 + 0x27cd0)
                                                    #19 0x00007f5f9f6f2d8a __libc_start_main (libc.so.6 + 0x27d8a)
                                                    #20 0x0000560aee6da465 _start (Xorg + 0x3f465)

                                                    Stack trace of thread 661:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 659:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 658:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 673:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 664:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 663:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 660:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 667:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 671:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 672:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 662:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 674:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 832:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 668:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 2766:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 676:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 675:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 677:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 2767:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 705:
                                                    #0  0x00007f5f9f754700 n/a (libc.so.6 + 0x89700)
                                                    #1  0x00007f5f9f75af1a pthread_mutex_lock (libc.so.6 + 0x8ff1a)
                                                    #2  0x0000560aee7eca35 n/a (Xorg + 0x151a35)
                                                    #3  0x0000560aee7ee1f2 n/a (Xorg + 0x1531f2)
                                                    #4  0x0000560aee7f1849 n/a (Xorg + 0x156849)
                                                    #5  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #6  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)
                                                    ELF object binary architecture: AMD x86-64
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null.monitor"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null.monitor"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_input.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp_6__source"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux systemd[1]: Started Process Core Dump (PID 62853/UID 0).
oct. 24 22:18:21 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:21 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:17 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:16 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:18:16 archlinux rtkit-daemon[873]: Supervising 1 threads of 1 processes of 1 users.
oct. 24 22:17:57 archlinux wpa_supplicant[533]: wlp0s20f3: CTRL-EVENT-SIGNAL-CHANGE above=1 signal=-59 noise=9999 txrate=26000
oct. 24 22:17:02 archlinux systemd[773]: kde-baloo.service: Consumed 6h 32min 53.481s CPU time.
oct. 24 22:17:02 archlinux systemd[773]: kde-baloo.service: Failed with result 'oom-kill'.
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 677:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 2767:
                                                    #0  0x00007f5f9f7544ae n/a (libc.so.6 + 0x894ae)
                                                    #1  0x00007f5f9f756d40 pthread_cond_wait (libc.so.6 + 0x8bd40)
                                                    #2  0x00007f5f9c6dab5c n/a (iris_dri.so + 0xd1b5c)
                                                    #3  0x00007f5f9c725a0c n/a (iris_dri.so + 0x11ca0c)
                                                    #4  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #5  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)

                                                    Stack trace of thread 705:
                                                    #0  0x00007f5f9f754700 n/a (libc.so.6 + 0x89700)
                                                    #1  0x00007f5f9f75af1a pthread_mutex_lock (libc.so.6 + 0x8ff1a)
                                                    #2  0x0000560aee7eca35 n/a (Xorg + 0x151a35)
                                                    #3  0x0000560aee7ee1f2 n/a (Xorg + 0x1531f2)
                                                    #4  0x0000560aee7f1849 n/a (Xorg + 0x156849)
                                                    #5  0x00007f5f9f7579eb n/a (libc.so.6 + 0x8c9eb)
                                                    #6  0x00007f5f9f7db7cc n/a (libc.so.6 + 0x1107cc)
                                                    ELF object binary architecture: AMD x86-64
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null.monitor"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null.monitor"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "auto_null"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_input.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp_6__source"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux plasmashell[954]: org.kde.plasma.pulseaudio: No object for name "alsa_output.pci-0000_00_1f.3-platform-skl_hda_dsp_generic.HiFi__hw_sofhdadsp__sink"
oct. 24 22:19:06 archlinux systemd[1]: Started Process Core Dump (PID 62853/UID 0).```

**Comment by pwouik** [comment]
caused by https://github.com/rust-windowing/winit/issues/3245

**Comment by pwouik** [comment]
fixed by 048c5103e27f62a5605f841d2915af14cd183565
